src/middleware/cache.rs

```diff
@@ -0,0 +1,48 @@
+use async_trait::async_trait;
+use jsonrpsee::core::{Error, JsonValue};
+
+use super::{Middleware, NextFn};
+use crate::{cache::Cache, middleware::call::CallRequest};
+
+pub struct CacheMiddleware {
+    cache: Cache,
+}
+
+impl CacheMiddleware {
+    pub fn new(cache: Cache) -> Self {
+        Self { cache }
+    }
+}
+
+#[async_trait]
+impl Middleware<CallRequest, Result<JsonValue, Error>> for CacheMiddleware {
+    async fn call(
+        &self,
+        request: CallRequest,
+        next: NextFn<CallRequest, Result<JsonValue, Error>>,
+    ) -> Result<JsonValue, Error> {
+        if request.skip_caching {
+            return next(request).await;
+        }
+
+        let params_joined = request
+            .params
+            .iter()
+            .map(|x| x.to_string())
+            .collect::<Vec<_>>()
+            .join(",");
+        let key = format!("{}::[{}]", request.method, params_joined);
```

Comment:
the key should be hash of params as params could be large
and using haser we don't need to join the paramers

Reply:
Lru has it's own hasher but still we can hash it